src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ZERO_INDEX = 0;
+    private static final int USUALLY_SIZE = 1;
+    private int capacity = DEFAULT_CAPACITY;
+    private int size;
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
+
+    public void checkSize(int length) {
+        while ((size == capacity) || (length + size > capacity)) {
+            capacity = (int) (capacity * 1.5 + 1);
+            Object [] inheritElementData = new Object[capacity];
+            System.arraycopy(elementData, 0, inheritElementData, 0, size);
+            elementData = new Object[capacity];
+            System.arraycopy(inheritElementData, 0, elementData, 0, size);
+        }
+    }
+
+    public void checkGetIndex(int index) {
+        if ((index >= size) || (index < ZERO_INDEX)) {
+            throw new ArrayListIndexOutOfBoundsException("Undefined index");
+        }
+    }
+
+    public void checkAddRemoveIndex(int index) {
```

Comment:
let's rename the method - will you be able to remove the element on index 5 if size is 5?